### PR TITLE
[Tools] Accelerate model loading.

### PR DESCRIPTION
**Pull request by marvin-Yu**
qwen 7b loading time from `24s` to `16s`;
qwen 72b loading time from `517s` to `400s`;

### CODE REVIEW

**Review by pujiang2018 — COMMENTED** [review]
1. need to be careful to add omp in float16.h, as the outer scope may already assume it is in simple thread, and contains omp at the outside scope.
2. can we remove filesystem linkage? (as current binary is already big, we may only add new library when it is a MUST)

**Comment by Duyi-Wang** [comment]
> 1. need to be careful to add omp in float16.h, as the outer scope may already assume it is in simple thread, and contains omp at the outside scope.
> 2. can we remove filesystem linkage? (as current binary is already big, we may only add new library when it is a MUST)

We have introduced filesystem in `debugger.h`

**Comment by pujiang2018** [comment]
> > 1. need to be careful to add omp in float16.h, as the outer scope may already assume it is in simple thread, and contains omp at the outside scope.
> > 2. can we remove filesystem linkage? (as current binary is already big, we may only add new library when it is a MUST)
> 
> We have introduced filesystem in `debugger.h`

there is a difference that debugger will ONLY be effective when DEBUG is enabled.

File: src/common/float16.h [hunk 1]
```diff
@@ -170,11 +202,33 @@ inline void float16_t::cvt_float16_to_float(const float16_t *src, float *dst, in
...
+        _mm512_mask_storeu_ps(dst + size - remainder, mask, output_vector);
+    }
+}
+
 inline void float16_t::float_add_float16(const float *src1, const float16_t *src2, float *dst, int size) {
     constexpr int kStep = 16;
     int blockSize = size / kStep;
     int remainder = size % kStep;
 
+#pragma omp parallel for
```
Comment (pujiang2018) [inline src/common/float16.h:231]: may have risk here. better not change.

Comment (marvin-Yu) [reply src/common/float16.h:231]: done

**Review by pujiang2018 — APPROVED** [review]

**Pull request merged**